scripts/helpers/function-helpers.ts

```diff
@@ -0,0 +1,135 @@
+import { Keyring } from "@polkadot/api";
+import { blake2AsHex } from "@polkadot/util-crypto";
+import { hexToU8a } from "@polkadot/util";
+
+export function schedulerWrapper(api, atBlock, tx) {
+  return api.tx.scheduler.schedule(atBlock, null, 0, tx);
+}
+
+export async function accountWrapper(api, privateKey) {
+  // Keyring
+  const keyring = new Keyring({ type: "ethereum" });
+
+  // Create account and get nonce
+  let account = await keyring.addFromUri(privateKey, null, "ethereum");
+  const { nonce: rawNonce } = (await api.query.system.account(account.address)) as any;
+  let nonce = BigInt(rawNonce.toString());
+
+  return [account, nonce];
+}
+
+export async function sudoWrapper(api, tx, account) {
+  try {
+    await api.tx.sudo.sudo(tx).signAndSend(account);
+    console.log("--> Sudo Tx sent\n");
+  } catch (e) {
+    console.log(e.message);
+  }
+}
+
+export async function preimageWrapper(api, tx, account, nonce) {
+  // Prepare the proposal hash
+  let preimage = {};
+  preimage["encodedProposal"] = tx?.method.toHex() || "";
+  preimage["encodedLength"] = hexToU8a(preimage["encodedProposal"]).length;
+  preimage["encodedHash"] = blake2AsHex(preimage["encodedProposal"]);
+  console.log("Encoded proposal hash is %s", preimage["encodedHash"]);
+  console.log("Encoded length %d", preimage["encodedLength"]);
+
+  // Preimage depending on RT version
+  let runtimeVersion = BigInt(api.runtimeVersion.specVersion);
+
+  // Check if preimage exists
+  let preimageCheck;
+  if (runtimeVersion < 2000n) {
+    preimageCheck = await api.query.democracy.preimages(preimage["encodedHash"]);
+  } else {
+    preimageCheck = await api.query.preimage.preimageFor([
+      preimage["encodedHash"],
+      preimage["encodedLength"],
+    ]);
+  }
+
+  // If preimage does not exists
+  if (preimageCheck.toHex() == "0x") {
```

Comment:
I think you should be able to do `if(preimate)`

Reply:
I tried that first, but when you query it from on-chain and there is nothing it returned `0x` so I had to force it to this 🤷‍♂️